File: university-service/src/main/java/ua/com/foxminded/courseproject/service/ScheduleService.java
package ua.com.foxminded.courseproject.service;

import ua.com.foxminded.courseproject.dto.ScheduleDto;

import java.util.List;
import java.util.UUID;

public interface ScheduleService {
    public ScheduleDto findById(UUID id);

    public List<ScheduleDto> findAll();
}

File: university-service/src/main/java/ua/com/foxminded/courseproject/service/ScheduleService.kt
package ua.com.foxminded.courseproject.service

import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ua.com.foxminded.courseproject.dto.ScheduleDto
import java.util.*

interface ScheduleService {
    fun findById(id: UUID): Mono<ScheduleDto>
    fun findAll(): Flux<ScheduleDto>
}